torchgeo/datasets/chesapeake.py

```diff
@@ -40,6 +40,7 @@ class Chesapeake(RasterDataset, abc.ABC):
     """
 
     is_image = False
+    dtype = torch.long
```

Comment:
I'm guessing we need to set this for all other current mask datasets?
```console
$ grep 'is_image = False' *.py
agb_live_woody_density.py:    is_image = False
astergdem.py:    is_image = False
cdl.py:    is_image = False
chesapeake.py:    is_image = False
cms_mangrove_canopy.py:    is_image = False
esri2020.py:    is_image = False
eudem.py:    is_image = False
globbiomass.py:    is_image = False
nlcd.py:    is_image = False
```

Reply:
only datasets where the dtype of the mask needs to be overridden

Reply:
> I'm fine with this approach. We could also use something like:
> 
> ```python
> @property
> def dtype(self) -> torch.dtype:
>     if self.is_image:
>         return torch.float32
>     else:
>         return torch.long
> ```
> 
> Then we would only need to override this for the small number of datasets that have a different dtype.

But the whole point is that we sometimes want to allow for non `torch.long` masks

Reply:
Yes, and adding `dtype = torch.float32` to a subclass will override this property. If `dtype` isn't set in a subclass, the property will be used instead.

Reply:
And then always cast to dtype in `__getitem__(...)`? Why is that better than what we're currently doing in this PR? 

Also, we could set this for the other mask datasets if we wanted to be explicit! Let me know.

Reply:
Many raster files are stored in uint8–uint32. Instead of adding `dtype = torch.long` or `dtype = torch.float32` to all of those datasets, we can set it once in the base class and override it only when a dataset needs to be different (none of our current datasets, but future pixel-wise regression datasets).